textdb/textdb-storage/src/main/java/edu/uci/ics/textdb/storage/LuceneDataStore.java

```diff
@@ -0,0 +1,87 @@
+package edu.uci.ics.textdb.storage;
+
+import java.io.IOException;
+import java.nio.file.Paths;
+import java.util.List;
+
+import org.apache.lucene.analysis.Analyzer;
+import org.apache.lucene.analysis.standard.StandardAnalyzer;
+import org.apache.lucene.document.Document;
+import org.apache.lucene.index.IndexWriter;
+import org.apache.lucene.index.IndexWriterConfig;
+import org.apache.lucene.store.Directory;
+import org.apache.lucene.store.FSDirectory;
+
+import edu.uci.ics.textdb.api.common.IField;
+import edu.uci.ics.textdb.api.common.ITuple;
+import edu.uci.ics.textdb.api.storage.IDataStore;
+import edu.uci.ics.textdb.common.field.Attribute;
+import edu.uci.ics.textdb.common.field.FieldType;
+import edu.uci.ics.textdb.common.utils.Utils;
+
+public class LuceneDataStore implements IDataStore{
+    
+    private String indexDir;
+
+    public LuceneDataStore(String indexDir) {
+        this.indexDir = indexDir;
+    }
+    
+    public void clearData() throws Exception{
+        IndexWriter indexWriter = null;
+        try {
+            Directory directory = FSDirectory.open(Paths
+                    .get(indexDir));
+            Analyzer analyzer = new StandardAnalyzer();
+            IndexWriterConfig conf = new IndexWriterConfig(analyzer);
+            indexWriter = new IndexWriter(directory, conf);
+            indexWriter.deleteAll();
+        } catch (Exception e) {
+            e.printStackTrace();
+            throw e;
+        } finally {
+            if (indexWriter != null) {
+                indexWriter.close();
+            }
+        }
+        
+    }
+    
+    public void storeData(List<Attribute> schema, List<ITuple> tuples) throws Exception {

```

Comment:
I also wanted pull the methods cleatDate() and storeData() to the interface, but storeData() method accepts List<Attribute> parameter. Currently Attribute is present in textdb-common (on which textdb-api shouldn't depend ).. So, we need to create an IAttribute interface in textdb-api and move FieldType enum class to textdb-api (Since IAttribute will have methods which use FieldType enum). Shall I make these changes?